quodlibet/util/path.py

```diff
@@ -50,6 +52,17 @@ def fsn2glib(path):
     return path
 
 
+def uri2gsturi(uri):
+    """Takes a correct uri and returns a gstreamer-compatible uri"""
+    if not is_windows():
+        return uri
+    try:
+        # gstreamer requires extra slashes for network shares
+        return GLib.filename_to_uri(uri2fsn(uri))
```

Comment:
Sorry, I missed that before,, but you also need to catch ValueError, in case non-fiel URIs get passed (like "http://etc..")

Reply:
I've fixed that and added some tests: 1a1284578e77a6